crates/engine/src/analysis/variable.rs

```diff
@@ -73,62 +68,28 @@ universal_id! {
     UVID => EDB_RUNTIME_VALUE_OFFSET
 }
 
-/// A reference-counted pointer to a Variable.
-///
-/// This type alias provides shared ownership of Variable instances, allowing
-/// multiple parts of the analysis system to reference the same variable
-/// without copying the data.
-#[derive(Clone, derive_more::Debug)]
-#[allow(unused)]
-pub struct VariableRef {
-    inner: Arc<RwLock<Variable>>,
-    /* cached readonly fields*/
-    #[debug(ignore)]
-    name: OnceCell<String>,
-    #[debug(ignore)]
-    declaration: OnceCell<VariableDeclaration>,
-}
-
-impl From<Variable> for VariableRef {
-    fn from(variable: Variable) -> Self {
-        Self::new(variable)
+define_ref! {
+    /// A reference-counted pointer to a Variable.
+    ///
+    /// This type alias provides shared ownership of Variable instances, allowing
+    /// multiple parts of the analysis system to reference the same variable
+    /// without copying the data.
+    #[allow(unused)]
+    VariableRef(Variable) {
+        cached_method: {
+            declaration: VariableDeclaration,
+            type_name: Option<TypeName>,
+        }
+        delegate: {
+            fn id(&self) -> UVID;
+            fn contract(&self) -> Option<ContractRef>;
+            fn function(&self) -> Option<FunctionRef>;
+        }
```

Comment:
I guess we will have another PR to distinguish parameters and return value?

Reply:
I will put them in another PR. This PR focuses on refactoring